x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/ClusterDeprecationChecks.java

```diff
@@ -290,4 +291,95 @@ static DeprecationIssue checkGeoShapeTemplates(final ClusterState clusterState)
             return null;
         }
     }
+
+    protected static boolean isSparseVector(Map<?, ?> property) {
+        return "sparse_vector".equals(property.get("type"));
+    }
+
+    protected static String formatDeprecatedSparseVectorMessage(String type, Map.Entry<?, ?> entry) {
+        return entry.getKey().toString();
+    }
+
+    @SuppressWarnings("unchecked")
+    private static String getDetailsMessageForSparseVectorComponentTemplates(Map<String, ComponentTemplate> componentTemplates) {
+        String detailsForComponentTemplates =
+            componentTemplates.entrySet().stream().map((templateCursor) -> {
+                String templateName = templateCursor.getKey();
+                ComponentTemplate componentTemplate = templateCursor.getValue();
+                CompressedXContent mappings = componentTemplate.template().mappings();
+                if (mappings != null) {
+                    Tuple<XContentType, Map<String, Object>> tuple = XContentHelper.convertToMap(mappings.uncompressed(), true,
+                        XContentType.JSON);
+                    Map<String, Object> mappingAsMap = tuple.v2();
+                    List<String> messages = mappingAsMap == null ? Collections.emptyList() :
+                        IndexDeprecationChecks.findInPropertiesRecursively(LegacyGeoShapeFieldMapper.CONTENT_TYPE,
+                            mappingAsMap,
+                            ClusterDeprecationChecks::isSparseVector,
+                            ClusterDeprecationChecks::formatDeprecatedSparseVectorMessage);
+                    if (messages.isEmpty() == false) {
+                        String messageForMapping =
+                            "mappings in component template " + templateName + " contains deprecated sparse_vector fields: " +
```

Comment:
lets surround the template name in some kind of quotations or punctuation so that it's easier to see as a resource name instead of just text